File: 구름 알고리즘 미션/재귀/Bigger than you/Bigger than you.py
import sys
sys.setrecursionlimit(10**6)  # 재귀 깊이 제한 늘리기

N = int(input())
A = list(map(int, input().split()))
K = int(input())

def find_result(cur: str, depth: int) -> None:
	global result
	
	if cur and int(cur) > K:
		result = min(result, int(cur))
		return
	
	if depth > 10:
		return
	
	for num in A:
		find_result(cur + str(num), depth + 1)
		
result = float('inf')
find_result("", 0)
print(result)

File: 구름 알고리즘 미션/재귀/Bigger than you/Bigger than you.js
// Run by Node.js
const readline = require('readline');

(async () => {
    let rl = readline.createInterface({ input: process.stdin });
    let input = [];
    for await (const line of rl) {
        input.push(line);
        if (input.length === 3) rl.close();
    }

    const N = +input[0];
    const A = input[1].split(" ").map(Number).sort((a, b) => a - b); // 작은 값부터 탐색
    const K = +input[2];
    
    let result = Infinity;

    function findResult(cur) {
				let num = cur.length > 0 ? +cur : 0;
				if (cur.length > 0 && cur[0] === '0') return;
				if (cur.length > String(K).length + 1) return;
			
        // 현재 숫자가 K보다 크다면 결과 갱신
        if (num > K) {
            result = Math.min(result, num);
            return;
        }

        for (let digit of A) {
            findResult(cur + digit);
        }
    }

    findResult("");
    console.log(result);
    process.exit();
})();
